aopy/visualization/base.py

```diff
@@ -2498,4 +2498,115 @@ def plot_plane(plane, ax=None, gain=1.0, color='grey', alpha=0.15, resolution=10
+            transparent. Must be between 0.4 and 1. Default is 1.
+        ax (mpl_toolkits.mplot3d.Axes3D, optional): The Matplotlib 3D axis on which to plot the sphere.
+
+    Raises:
+        ValueError: If the 'alpha' value is outside the valid range [0.4, 1].
+
+    Examples:
+        To plot a semi-transparent blue sphere with a radius of 1 at the origin:
+
+        .. code-block:: python
+
+            import matplotlib.pyplot as plt
+            from mpl_toolkits.mplot3d import Axes3D
+
+            fig = plt.figure()
+            ax = fig.add_subplot(111, projection='3d')
+            plot_sphere(location=(0, 1, 2), color='blue', radius=5, resolution=30, alpha=0.5, ax=ax)
+
+        .. image:: _images/plot_sphere_example.png
+    """
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection='3d')
+    
+    # Assert that alpha is within the valid range [0.4, 1]
+    if not (0.4 <= alpha <= 1):
+        raise ValueError(f"Invalid value for transparency (alpha): {alpha}. Must be between 0.4 and 1.")
+    
+    # Generate points in spherical coordinates
+    phi = np.linspace(0, 2 * np.pi, resolution) # azimuthal angle
+    theta = np.linspace(0, np.pi, resolution) # polar angle
+    
+    # Translate to cartesian coordinates
+    x = radius * np.outer(np.cos(phi), np.sin(theta)) + location[0]
+    y = radius * np.outer(np.sin(phi), np.sin(theta)) + location[1]
+    z = radius * np.outer(np.ones(np.size(phi)), np.cos(theta)) + location[2]
+    
+    # Plot sphere
+    ax.plot_surface(x, y, z, color=color, alpha=alpha-0.4)
+
+def plot_targets_3D(target_locations, colors, target_radius=1, resolution=20, alpha=1, ax=None, **kwargs):
+    """
+    Plots multiple targets as spheres in 3D space.
+
+    Args:
+        target_locations (list of tuples or lists): List of (x, y, z) coordinates specifying the 
+            centers of the target spheres.
+        colors (list of str or None, optional): List of colors for the targets. If not provided, all 
+            targets will default to black. Must match the number of unique targets.
+        target_radius (float, optional): Radius of each target sphere. Default is 1.
+        resolution (int, optional): Resolution of the spheres (passed to 'plot_sphere'). Default is 20.
+        alpha (float, optional): Transparency of the spheres, where 1 is opaque, and 0.4 is the minimum 
+            allowed transparency. Default is 1.
+        ax (mpl_toolkits.mplot3d.Axes3D, optional): The Matplotlib 3D axis on which to plot the targets.
+
+    Raises:
+        ValueError: If 'alpha' is outside the range [0.4, 1].
+        ValueError: If 'colors' is less than the number of unique targets.
+
+    Notes:
+        - This function calls 'plot_sphere' for each target in 'target_locations'.
+        - If 'colors' is not specified, all targets will be rendered in black.
+        - The number of colors provided must be at least equal to the number of unique target locations.
```

Comment:
i think these notes are fully redundant with the args section and can be removed